File: app/src/main/java/empire/stark/firststep/AppComponent.java
package empire.stark.firststep;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;
import empire.stark.firststep.common.dagger.BaseActivityModule;
import empire.stark.firststep.common.dagger.BaseFragmentModule;

/**
 * Created by YEN_MINH on 3/8/2017.
 */
@Singleton
@Component(modules =
        {
                AppModule.class,
                AndroidSupportInjectionModule.class,
                BaseActivityModule.class,
                BaseFragmentModule.class
        }
)
public interface AppComponent {

    void inject(App app);

    //exported for child-component
    Application application();
}

File: app/src/main/java/empire/stark/firststep/AppComponent.kt
package empire.stark.firststep

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import dagger.android.support.DaggerApplication
import javax.inject.Singleton

/**
 * Created by YEN_MINH on 3/8/2017.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class, AndroidSupportInjectionModule::class))
interface AppComponent : AndroidInjector<DaggerApplication> {
    fun inject(app: App)

    override fun inject(instance: DaggerApplication?)

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: App): AppComponent.Builder

        fun build(): AppComponent
    }

}
